support/crown/src/unrooted_must_root.rs

```diff
@@ -263,16 +317,57 @@ impl<'tcx> LateLintPass<'tcx> for UnrootedPass {
                     .associated_items(impl_def_id)
                     .find_by_name_and_kind(cx.tcx, trait_item.ident, ty::AssocKind::Type, trait_id)
                     .unwrap();
+
                 let mir_ty = cx.tcx.type_of(type_impl.def_id).skip_binder();
-                if is_unrooted_ty(&self.symbols, cx, mir_ty, false) {
+
+                let impl_ty_must_root = associated_type_has_attr(sym, cx, mir_ty, sym.must_root);
+                let impl_ty_allow_unrooted =
+                    associated_type_has_attr(sym, cx, mir_ty, sym.allow_unrooted_interior);
+                let impl_ty_allow_rc =
+                    associated_type_has_attr(sym, cx, mir_ty, sym.allow_unrooted_in_rc);
+
+                if impl_ty_must_root != must_root_present {
+                    if !must_root_present && impl_ty_must_root {
+                        cx.lint(UNROOTED_MUST_ROOT, |lint| {
+                            lint.primary_message(
+                                "Type trait declaration must be marked with \
+                                 #[crown::unrooted_must_root_lint::must_root] \
+                                 to allow binding must_root types in associated types",
```

Comment:
Can we end all messages with a dot because rust inserts same text after it an makes it confusing.